### DISCUSSION TITLE
Programatically register handlers (instead of file based routing)?

### DISCUSSION THREAD

**Discussion by francisashley** [nitrojs/nitro#1089] [Q&A]
Hello, in the docs I can see it's possible to register handler functions directly in `devHandlers` https://nitro.unjs.io/config#devhandlers, however the feature is disabled in production ??. This feature is useful when the API file cannot return `defineEventHandler` as the default export.

Is there another way of registering handlers directly or perhaps this can be enabled in production on a future release?

**Comment by ananthachetan** [comment]
Take a look at this option: https://nitro.unjs.io/config#handlers

**Reply by francisashley** [reply]
Hi Anantha, I'm aware of this option.

The problem with the option is that it only supports passing a path to a file containing the handler (file based routing).

Here I am looking to register the raw function directly in the same way `devHandlers` behaves, e.g.

```
✅
'nitro:config'(nitroConfig) {
  registry.apps.forEach((app) => {
    nitroConfig.handlers?.push({
      method: 'get',
      route: 'api/some/endpoint',
      handler: defineEventHandler(event => { <------- like this
            // something
      })
    })
  })
}
```

```
❌
'nitro:config'(nitroConfig) {
  registry.apps.forEach((app) => {
    nitroConfig.handlers?.push({
      method: 'get',
      route: 'api/some/endpoint',
      handler: 'path/to/handler/file' <------- not like this
    })
  })
}
```
